### PR TITLE
Improve handle sessions

### PR DESCRIPTION
**Pull request by matthias-ronge**
- move logout handling from version management to session management (clean-up)
- log warnings if logout fails

### CODE REVIEW

**Comment by matthias-ronge** [comment]
Doesn’t work

File: Kitodo/src/main/java/org/kitodo/production/security/CustomLogoutSuccessHandler.java [hunk 1]
```diff
@@ -41,12 +46,21 @@ public CustomLogoutSuccessHandler(String onSuccessUrl) {
...
-                UserDetails user = (UserDetails) principal;
-                ServiceManager.getSessionService().expireSessionsOfUser(user);
+        if (Objects.nonNull(authentication)) {
+            if (Objects.nonNull(authentication.getDetails())) {
+                Object principal = authentication.getPrincipal();
+                if (principal instanceof UserDetails) {
+                    UserDetails user = (UserDetails) principal;
+                    ServiceManager.getSessionService().expireSessionsOfUser(user);
+                } else {
+                    logger.warn(MessageFormat.format("Cannot expire session: {0} !instanceof UserDetails",
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/security/CustomLogoutSuccessHandler.java:56]: I am not a friend of this code like syntax in natual lanaguage logs. Can we change this to "{0}  is not an instance of 'UserDetails'" or something similar? (the `!` is easy to overlook in a log message, in my opinion)

File: Kitodo/src/main/java/org/kitodo/production/services/security/SessionService.java [hunk 2]
```diff
@@ -37,6 +46,27 @@ private SessionService() {
...
+    @Override
+    public void sessionDestroyed(HttpSessionEvent se) {
+        Object securityContextObject = se.getSession().getAttribute("SPRING_SECURITY_CONTEXT");
+        if (securityContextObject instanceof SecurityContextImpl) {
+            SecurityContextImpl securityContext = (SecurityContextImpl) securityContextObject;
+            Object principal = securityContext.getAuthentication().getPrincipal();
+            if (principal instanceof SecurityUserDetails) {
+                expireSessionsOfUser((SecurityUserDetails) principal);
+            } else {
+                logger.warn(MessageFormat.format("Cannot expire session: {0} !instanceof SecurityUserDetails",
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/security/SessionService.java:61]: See above.

File: Kitodo/src/main/java/org/kitodo/production/services/security/SessionService.java [hunk 3]
```diff
@@ -37,6 +46,27 @@ private SessionService() {
...
+                    Helper.getObjectDescription(principal)));
+            }
+        } else {
+            logger.warn(MessageFormat.format("Cannot expire session: {0} !instanceof SecurityContextImpl",
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/services/security/SessionService.java:65]: See above.

**Review by solth — CHANGES_REQUESTED** [review]
@matthias-ronge thanks for updating the log messages. Please rebase your branch against the master and resolve the current merge conflict.

**Review by solth — APPROVED** [review]
Thanks for resolving the merge conflict.

**Pull request merged**